backend/src/controllers/person.controller.ts

```diff
@@ -122,57 +122,57 @@ export const deletePersons = async (
   req: Request,
   res: Response,
   next: NextFunction,
-  ): Promise<void> => {
-    logger.info("DELETE /persons/:personID request from frontend");
-    const auth_id = req.headers.authorization?.["user_id"];
-    
-    const current_user = await userService.getUserByAuthId(auth_id);
-    const id = req.params.id;
-
-    const user_persons = current_user?.persons;
-    let string_persons = user_persons?.map(x => x.toString());
-
-    if (string_persons?.includes(id.toString())) {
-      try {
-        // Delete person from database
-        const deletePersonsResult = await personService.deletePersons(id);
-
-        // return encounters that may have empty persons fields
-        const deleteEncountersResult = await encounterService.deleteEncounterPerson(id);
-        const empty_encounters = deleteEncountersResult["array"];
-        const EncountersBool = deleteEncountersResult["bool"];
-
-        //delete person from current User document
-        const deleteUserResult = await userService.deleteUserPerson(id);
-
-        // Check all service function calls were valid
-        if (EncountersBool && deletePersonsResult && deleteUserResult) {
-
-          // Make sure that empty encounters are also deleted from User
-          for (let i = 0; i < empty_encounters.length; i++) {
-            let result = await userService.deleteUserEncounter(empty_encounters[i]?._id.toString());
-
-            // Check that a valid deleteUserEncounter was executed
-            if (!result) {
-              res.sendStatus(httpStatus.BAD_REQUEST).end();
-            }
+): Promise<void> => {
+  logger.info('DELETE /persons/:personID request from frontend');
+  const authId = req.headers.authorization?.['user_id'];
+
+  const user = await userService.getUserByAuthId(authId);
+  const id = req.params.id;
+
+  if (!user) {
+    res.status(httpStatus.UNAUTHORIZED).end();
+  }
+
+  const userPersons = user?.persons;
+  const stringPersons = userPersons?.map((x) => x.toString());
+
+  if (stringPersons?.includes(id.toString())) {
+    try {
+      // Delete person from database
+      const deletePersonsResult = await personService.deletePersons(id);
+
+      // Return encounters that may have empty persons fields
+      const deleteEncountersResult = await encounterService.deleteEncounterPerson(id);
+      const emptyEncounters = deleteEncountersResult['array'];
+      const encountersBool = deleteEncountersResult['bool'];
+
+      // Delete person from current User document
+      const deleteUserResult = await userService.deleteUserPerson(id);
+
+      // Check all service function calls were valid
+      if (encountersBool && deletePersonsResult && deleteUserResult) {
+        // Make sure that empty encounters are also deleted from User
+        for (let i = 0; i < emptyEncounters.length; i + 1) {
```

Comment:
Not sure why but changing the i++ to i+1 seems to be causing some issues when attempting to delete multiple people from an encounter.

Reply:
okay, will change back